File: fix_all_syntax.py
#!/usr/bin/env python3
"""
Comprehensive script to fix all remaining syntax errors
"""
import os
import re
import glob
def fix_file_syntax(content):
    """Fix all common syntax errors in a file"""
    lines = content.split('\n')
    fixed_lines = []
    for line in lines:
        # Remove extra semicolons and quotes
        line = re.sub(r'";$', '";', line)
        line = re.sub(r'": value$', '', line)
        line = re.sub(r';\s*$', ';', line)
        # Fix unterminated strings
        if line.count('"') % 2 == 1 and not line.strip().endswith('"'):
            if 'import' in line or 'from' in line or '=' in line:
                line = line.rstrip() + '";'
            else:
                line = line.rstrip() + '"'
        # Fix malformed JSX
        line = re.sub(r'<([^>]+)\s*$', r'<\1>', line)
        # Remove standalone semicolons
        if line.strip() == ';':
            continue
        # Fix malformed imports
        if line.strip().startswith('import ') and not line.strip().endswith(';'):
            if '{' in line and '}' in line:
                line = line.rstrip() + ';'
            elif 'from' in line:
                line = line.rstrip() + ';'
        fixed_lines.append(line)
    return '\n'.join(fixed_lines)
def clean_merge_conflicts(content):
    """Clean merge conflict markers"""
    content = re.sub(r'<<<<<<<.*?\n', '', content, flags=re.DOTALL)
    content = re.sub(r'=======.*?\n', '', content, flags=re.DOTALL)
    content = re.sub(r'>>>>>>>.*?\n', '', content, flags=re.DOTALL)
    return content
def process_file(file_path):
    """Process a single file"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        original_content = content
        # Clean merge conflicts first
        content = clean_merge_conflicts(content)
        # Fix syntax errors
        content = fix_file_syntax(content)
        # Only write if content changed
        if content != original_content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            return True
        else:
            return False
    except Exception as e:
        print(f"Error processing {file_path}: {e}")
        return False
def main():
    """Main function to process all TypeScript and JavaScript files"""
    patterns = [
        'app/**/*.tsx',
        'app/**/*.ts',
        'components/**/*.tsx',
        'components/**/*.ts',
        'src/**/*.tsx',
        'src/**/*.ts',
        'pages/**/*.tsx',
        'pages/**/*.ts'
    ]
    files_processed = 0
    files_fixed = 0
    for pattern in patterns:
        for file_path in glob.glob(pattern, recursive=True):
            if 'node_modules' in file_path or 'dist' in file_path:
                continue
            files_processed += 1
            if process_file(file_path):
                files_fixed += 1
                print(f"Fixed: {file_path}")
    print(f"\nProcessed {files_processed} files")
    print(f"Fixed {files_fixed} files")
if __name__ == "__main__":
    main()

File: fix-all-syntax.js
const fs = require('fs');
const path = require('path');

// Function to fix syntax errors
function fixSyntaxErrors(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    let modified = false;

    // Fix 1: Fix malformed metadata exports
    content = content.replace(/export const metadata="Metadata "=/g, 'export const metadata: Metadata =');
    content = content.replace(/title="'/g, 'title: \'');
    content = content.replace(/description: '/g, 'description: \'');
    content = content.replace(/keywords: '/g, 'keywords: \'');
    content = content.replace(/openGraph: {/g, 'openGraph: {');
    content = content.replace(/type: '/g, 'type: \'');
    content = content.replace(/url: '/g, 'url: \'');

    // Fix 2: Fix malformed function declarations
    content = content.replace(/const (\w+)="React\.FC "=/g, 'const $1: React.FC =');
    content = content.replace(/const (\w+)="React\.FC"=/g, 'const $1: React.FC =');

    // Fix 3: Fix malformed object properties
    content = content.replace(/(\w+)="(\w+)"/g, '$1: $2');
    content = content.replace(/(\w+):\s*"(\w+)"/g, '$1: \'$2\'');

    // Fix 4: Fix unterminated strings
    content = content.replace(/'([^']*)\n([^']*)'/g, "'$1$2'");
    content = content.replace(/"([^"]*)\n([^"]*)"/g, '"$1$2"');

    // Fix 5: Fix malformed JSX attributes
    content = content.replace(/(\w+)="([^"]*)"([^"]*)"([^"]*)"/g, '$1="$2$3$4"');

    // Fix 6: Fix missing semicolons
    content = content.replace(/(\w+)\nimport/g, '$1;\nimport');
    content = content.replace(/(\w+)\nexport/g, '$1;\nexport');

    // Fix 7: Fix malformed imports
    content = content.replace(/import { ([^}]+) } from '([^']+)'\n([^;])/g, "import { $1 } from '$2';\n$3");

    if (modified || content !== fs.readFileSync(filePath, 'utf8')) {
      fs.writeFileSync(filePath, content);
      console.log(`Fixed: ${filePath}`);
      return true;
    }
    
    return false;
  } catch (error) {
    console.error(`Error fixing ${filePath}:`, error.message);
    return false;
  }
}

// Function to find all .tsx and .ts files
function findFiles(dir, extensions = ['.tsx', '.ts']) {
  let results = [];
  const list = fs.readdirSync(dir);
  
  list.forEach(file => {
    const filePath = path.join(dir, file);
    const stat = fs.statSync(filePath);
    
    if (stat && stat.isDirectory()) {
      if (file !== 'node_modules' && file !== '.next' && !file.startsWith('.')) {
        results = results.concat(findFiles(filePath, extensions));
      }
    } else {
      const ext = path.extname(file);
      if (extensions.includes(ext)) {
        results.push(filePath);
      }
    }
  });
  
  return results;
}

// Main execution
console.log('Starting comprehensive syntax fix...');

const appDir = path.join(__dirname, 'app');
const pagesDir = path.join(__dirname, 'pages');
const files = [...findFiles(appDir), ...findFiles(pagesDir)];

let fixedCount = 0;
let errorCount = 0;

files.forEach(file => {
  try {
    if (fixSyntaxErrors(file)) {
      fixedCount++;
    }
  } catch (error) {
    console.error(`Failed to process ${file}:`, error.message);
    errorCount++;
  }
});

console.log(`\nSyntax fix complete!`);
console.log(`Files fixed: ${fixedCount}`);
console.log(`Errors encountered: ${errorCount}`);
console.log(`Total files processed: ${files.length}`);
